sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/implementation/encryption/EncryptionItemRequestOptions.java

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+/**
+ *
+ * The {@link com.azure.cosmos.models.CosmosItemRequestOptions} that allows to specify options for encryption / decryption.
+ */
+package com.azure.cosmos.implementation.encryption;
+
+import com.azure.cosmos.implementation.encryption.api.EncryptionOptions;
+import com.azure.cosmos.models.CosmosItemRequestOptions;
+
+import java.util.function.Consumer;
+
+// TODO: how invoking setter methods chaining works? parent setters returns CosmosItemRequestOptions
+// this class setter require EncryptionItemRequestOptions
+public class EncryptionItemRequestOptions extends CosmosItemRequestOptions {
```

Comment:
What is the use case of inheritance here?

Reply:
The use case is to use the same method signature for `container.createItem(item, options)` and allow passing the encryption option.

Please see the .net version doing the same thing, we are trying to be in par with dotnet api:
https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/Microsoft.Azure.Cosmos.Encryption/src/EncryptionItemRequestOptions.cs#L12